snippets/base/forms.py

```diff
@@ -723,3 +725,35 @@ def save(self, *args, **kwargs):
         self.instance.jexl_expr = ' && '.join([x for x in jexl_expr_array if x])
 
         return super().save(*args, **kwargs)
+
+
+class AddonAdminForm(forms.ModelForm):
+    def clean(self, *args, **kwargs):
+        cleaned_data = super().clean(*args, **kwargs)
+        if 'url' in self.changed_data:
+
+            # Add ending slash and remove query params when needed.
+            url = cleaned_data.pop('url')
+            if not url.endswith('/'):
+                url += '/'
+            url = url[:-(len(urlparse(url).query) + 1)]
```

Comment:
Should these steps be reversed? Currently, if you have a URL with a querystring, a trailing slash gets added after the querystring (e.g. `?foo=bar&boom=bap/`), then the query string _and_ the trailing slash get stripped by the `urlparse` line.

Reply:
Great catch! I updated the code to handle this correctly and also removed the ending slash check from the other method